lib/internal/fs/utils.js

```diff
@@ -718,12 +729,28 @@ const validateRmOptionsSync = hideStackFrames((path, options) => {
       throw err;
     } else if (err.code === 'ENOENT' && !options.force) {
       throw err;
+    } else if (warn && err.code === 'ENOENT') {
+      emitPermissiveRmdirWarning();
     }
   }
 
   return options;
 });
 
+let permissiveRmdirWarned = false;
+
+function emitPermissiveRmdirWarning() {
+  if (!permissiveRmdirWarned) {
+    process.emitWarning(
+      'Permissive rmdir recursive is deprecated, use rm recursive and force \
+instead',
```

Comment:
Non-blocking question (as we can always change the text later) but would it be more clear to the typical end user if we duplicate the text from deprecations.md above? I don't think the typical user is going to know what _permissive_ means in this context.

```suggestion
      'In future versions of Node.js, fs.rmdir(path, { recursive: true }) will ' +
      'throw if path does not exist or is a file. Use fs.rm(path, { recursive: ' +
      'true, force: true }) instead',
```

Reply:
Good call. I need to update the tests to match this message so just updating my PR now.